### PR TITLE
Try to split player service (To review)

### PR DESCRIPTION
**Pull request by LeaklessGfy**
Short Description: 
Try to make services inside this mess (first incrementation)

### CODE REVIEW

**Comment by douglascamata** [comment]
Merge conflicts :/

**Comment by LeaklessGfy** [comment]
@douglascamata fix

**Comment by LeaklessGfy** [comment]
Got other stuff to fix

File: pokemongo_bot/services/player_service.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    def __init__(self, api, config):
+        self.api = api
+        self.config = config
+        self.latest_inventory = None
+
+    def update_inventory(self):
+        response = self.get_inventory()
+        inventory = list()
+
+        if not 'responses' in response:
```
Comment (TheSavior) [inline pokemongo_bot/services/player_service.py]: can't this whole thing be
```
r = response.get('responses', {}).get('GET_INVENTORY', {}).get('inventory_delta', {}).get('inventory_items', {})
```

**Comment by surfaace** [comment]
what was the 'fix' commit a4c7426df90c9617bee377d5d0ef0e6dd110b2d3 ?

**Comment by douglascamata** [comment]
@LeaklessGfy you mixed a lot of fixes with code move and we're a bit afraid to merge this. Can you organise commits in a better way? Maybe split this PR in 2 (or more): 

1. Just code extraction from bot's main class
2. Fixes to the extracted code

**Comment by jtdroste** [comment]
Please use the `.get` method on dicts, rather than checking every nested level and returning if it doesn't exist.  This extends @TheSavior's comment - however there's two other areas that are just like it that I'd like you to change.

In the future *PLEASE* title your commits appropriately.  Having a commit message say "Fix" gives us NO information what was changed

File: pokemongo_bot/services/player_service.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        user_web_inventory = 'web/inventory-%s.json' % (self.config.username)
+        with open(user_web_inventory, 'w') as outfile:
+            json.dump(inventory_dict, outfile)
+
+        # get player items stock
+        items_stock = {x.value:0 for x in list(Item)}
+
+        for item in inventory_dict:
+            try:
+                # print(item['inventory_item_data']['item'])
```
Comment (bstpierr) [inline pokemongo_bot/services/player_service.py]: Please remove all commented-out code.

File: pokemongo_bot/services/player_service.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+        if response_dict:
+            self._player = response_dict['responses']['GET_PLAYER']['player_data']
+            player = self._player
+        else:
+            logger.log("The API didn't return player info, servers are unstable - retrying.", 'red')
+            sleep(5)
+            self.print_character_info()
+
+        # @@@ TODO: Convert this to d/m/Y H:M:S
```
Comment (bstpierr) [inline pokemongo_bot/services/player_service.py]: Did you mean to do this before merging the PR?

File: pokemongo_bot/services/player_service.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+            ' | IncenseCool: ' + str(items_stock[403]), 'cyan')
+        logger.log('Revive: ' + str(items_stock[201]) +
+            ' | MaxRevive: ' + str(items_stock[202]), 'cyan')
+
+        logger.log('')
+
+    def get_inventory_count(self, what):
+        response_dict = self.get_inventory()
+
+        if not 'responses' in response_dict:
```
Comment (bstpierr) [inline pokemongo_bot/services/player_service.py]: Same comment here about using `get` with `{}` as the default.

**Comment by LeaklessGfy** [comment]
All the comment are pretty interesting because I just move code on this PR. No new function so these comment concern former dev' not me especially. But I can do that of curse.

This merge is important because it's the first step to get thing more modular and structured.

File: pokemongo_bot/services/player_service.py (hunk 3 above)
Comment (LeaklessGfy) [inline pokemongo_bot/services/player_service.py]: What you mean ?

**Comment by TheSavior** [comment]
@LeaklessGfy I think @douglascamata's last comment represents accurately where the dev team feels about this:

```
@LeaklessGfy you mixed a lot of fixes with code move and we're a bit afraid to merge this. Can you organise commits in a better way? Maybe split this PR in 2 (or more):

Just code extraction from bot's main class
Fixes to the extracted code
```

We are scared of this large of a refactor. It's too hard to feel confident in it from looking at the diff. If there is any way to split it up into smaller PRs, it would be beneficial to all of us and they would get merged quicker.

File: pokemongo_bot/services/player_service.py (hunk 3 above)
Comment (LeaklessGfy) [inline pokemongo_bot/services/player_service.py]: The todo comment is not mine ;)

**Comment by TheSavior** [comment]
This PR is really far behind dev and difficult to review properly. We would still love to split this file up but would prefer doing it in many smaller, easy to understand pull requests. 

@LeaklessGfy, if you are still interested in doing that, it would be much appreciated but we would understand if you don't.

**Pull request closed without merging**